### ISSUE TITLE
Syntax coloring partially missing for JavaScript blocks inside HTML file

### ISSUE THREAD

**Issue by UweKeim** [microsoft/vscode#205942]
Does this issue occur when all extensions are disabled?: **Yes**

(I've even removed _all_ user settings from "C:\Users\ukeim\AppData\Roaming\Code\User\settings.json" and restarted VS Code with no improvement)

- VS Code Version: 1.86.2 (user setup)
- OS Version: Windows_NT x64 10.0.22631

Steps to Reproduce:

1. Open [this attached HTML file](https://github.com/microsoft/vscode/files/14370531/zp-internal-preview.zip) into VS Code.
2. See that syntax coloring is not applied to all JavaScript code inside the `<script>` block.

Please note that for a short amount of time (approx. half a second) right after opening the file in VS Code, the syntax coloring is correct and then flashes (and stays) immediately to wrong.

Example screenshots of incomplete syntax coloring:

![image](https://github.com/microsoft/vscode/assets/415439/d6f2fec5-3c1b-4b78-bfdf-ccab2e3f1a31)

![image](https://github.com/microsoft/vscode/assets/415439/35d16211-e7f1-4765-870a-f1e27311ecf4)

(I've underlined some, but not all, items missing color)

**Comment by UweKeim** [comment]
OMG, I figured out the reason.

My attached HTML file has this in line 152:

```html
<script>var $z = $.noConflict(true); var showFrameTimeout</script>
```

When removing this line, the whole HTML syntax-coloring is correct, again.

After digging further, I found out that the missing semicolon (`;`) right after `showFrameTimeout` is the cause of the whole mess.

To fix this, I rewrote the line 152 this way:

```html
<script>var $z = $.noConflict(true); var showFrameTimeout;</script>
```

After that, everything looks correctly colored.

This was the most important semicolon in my whole developer's life of over 25 years 🥰.

Still, I would _love_ if VS Code would not stumble (and fall) over one missing semicolon.
